apiserver/facades/controller/firewaller/firewaller_test.go

```diff
@@ -104,19 +101,34 @@ func (s *firewallerSuite) TestGetAssignedMachine(c *gc.C) {
 
 func (s *firewallerSuite) openPorts(c *gc.C) {
 	// Open some ports on the units.
-	s.AssertOpenUnitPorts(c, s.units[0], s.subnet.ID(), "tcp", 1234, 1400)
-	s.AssertOpenUnitPort(c, s.units[0], "", "tcp", 4321)
-	s.AssertOpenUnitPorts(c, s.units[2], "", "udp", 1111, 2222)
+	allEndpoints := ""
+	s.mustOpenPorts(c, s.units[0], allEndpoints, []network.PortRange{
+		network.MustParsePortRange("1234-1400/tcp"),
+		network.MustParsePortRange("4321/tcp"),
+	})
+	s.mustOpenPorts(c, s.units[2], allEndpoints, []network.PortRange{
+		network.MustParsePortRange("1111-2222/udp"),
+	})
+}
+
+func (s *firewallerSuite) mustOpenPorts(c *gc.C, unit *state.Unit, endpointName string, portRanges []network.PortRange) {
+	unitPortRanges, portChangesFn, err := unit.OpenedPortRanges()
+	c.Assert(err, jc.ErrorIsNil)
+
+	for _, pr := range portRanges {
+		unitPortRanges.Open(endpointName, pr)
+	}
+
+	c.Assert(s.State.ApplyOperation(portChangesFn()), jc.ErrorIsNil)
```

Comment:
After reading code between here and the previous comment, perhaps a method called `Changes`?

Reply:
Addressed in https://github.com/juju/juju/pull/11863/commits/e3ab5027dca6cd28af2070b0107ef831fdae1a6e